crates/roc_host/src/json_web_token.rs

```diff
@@ -0,0 +1,170 @@
+use hmac::{Hmac, Mac};
+use jwt::algorithm::AlgorithmType;
+use roc_std::{RocList, RocResult, RocStr};
+use sha2::{Sha256, Sha384, Sha512};
+use std::collections::BTreeMap;
+
+#[derive(Clone, Default, Debug, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[repr(C)]
+pub struct JWTFromRoc {
+    secret: RocStr,
+    token: RocStr,
+    algo: u8,
+}
+
+impl JWTFromRoc {
+    fn algorithm(&self) -> AlgorithmType {
+        match self.algo {
+            1 => AlgorithmType::Hs256,
+            2 => AlgorithmType::Hs384,
+            3 => AlgorithmType::Hs512,
+            4 => AlgorithmType::Rs256,
+            5 => AlgorithmType::Rs384,
+            6 => AlgorithmType::Rs512,
+            7 => AlgorithmType::Es256,
+            8 => AlgorithmType::Es384,
+            9 => AlgorithmType::Es512,
+            10 => AlgorithmType::Ps256,
+            11 => AlgorithmType::Ps384,
+            12 => AlgorithmType::Ps512,
+            13 => AlgorithmType::None,
```

Comment:
Any reason glue can generate an enum for this instead of us dealing with raw integers?

Reply:
Done